### ISSUE TITLE
ansible 2.2.0.0 vs. 2.2.1.0 template error while templating string: unexpected '.'

### ISSUE THREAD

**Issue by chrismeyersfsu** [ansible/ansible#21782]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
* template + register + jinja2

##### ANSIBLE VERSION

```
ansible 2.1.4.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

**ansible 2.2.1.0 is the version with the change in behavior; error**
```
ansible 2.2.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

Regression or change in behavior from Ansible 2.2.0.0 to Ansible 2.2.1.0 when using a registered variable with content in it that looks like a jinja2 template.

##### STEPS TO REPRODUCE

How I arrived at the below example recreation playbook stems from my use case.

```
# command line
docker inspect --format='{{.NetworkSettings.IPAddress}}' container_here

# Ansibilized: 
shell: "docker inspect --format='{{.NetworkSettings.IPAddress}}' container_here"

# Ansibilized + jinja friendly: 
shell: "docker inspect --format '{{ '{{' }} .NetworkSettings.IPAddress {{ '}}' }}' container_here"

# Ansibilized + jinja friendly + use the result in ansible
shell: "docker inspect --format '{{ '{{' }} .NetworkSettings.IPAddress {{ '}}' }}' container_here"
register: results
```

Note that I don't care about the full `{{ results }}` variable. I only really want the `{{ results.stdout }}`. However, the args from `shell` module invocation are included in the `{{ results.cmd }}` variable .. and that is problematic.

```
- hosts: localhost
  tasks:
    - local_action:
        module: "shell"
        args: "echo '{{ '{{' }} .NetworkSettings.IPAddress {{ '}}' }}' > /dev/null"
        #args: "echo '{{ .NetworkSettings.IPAddress }}'"
      register: result

    - debug:
        msg: "{{ result.stdout }}"

    - debug:
        msg: "{{ result }}"
```

##### EXPECTED RESULTS

```
(ansible_2.1.4.0) meyers@macbeefy:~/ansible/ansible_bugs/args_template$ $ ansible-playbook main.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [command] *****************************************************************
changed: [localhost -> localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": ""
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": {
        "changed": true,
        "cmd": "echo '{{ .NetworkSettings.IPAddress }}' > /dev/null",
        "delta": "0:00:00.009223",
        "end": "2017-02-22 10:29:30.495296",
        "rc": 0,
        "start": "2017-02-22 10:29:30.486073",
        "stderr": "",
        "stdout": "",
        "stdout_lines": [],
        "warnings": []
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=1    unreachable=0    failed=0
```

```
(ansible_2.2.0.0) meyers@macbeefy:~/ansible/ansible_bugs/args_template$ $ ansible-playbook main.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [command] *****************************************************************
changed: [localhost -> localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": ""
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": {
        "changed": true,
        "cmd": "echo '{# .NetworkSettings.IPAddress #}' > /dev/null",
        "delta": "0:00:00.009168",
        "end": "2017-02-22 10:30:03.330081",
        "rc": 0,
        "start": "2017-02-22 10:30:03.320913",
        "stderr": "",
        "stdout": "",
        "stdout_lines": [],
        "warnings": []
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=1    unreachable=0    failed=0
```

##### ACTUAL RESULTS

```
(ansible_2.2.1.0) meyers@macbeefy:~/ansible/ansible_bugs/args_template$ $ ansible-playbook main.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [command] *****************************************************************
changed: [localhost -> localhost]

TASK [debug] *******************************************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "{u'cmd': u\"echo '{{ .NetworkSettings.IPAddress }}' > /dev/null\", u'end': u'2017-02-22 10:30:27.539643', u'changed': True, u'stdout': u'', u'start': u'2017-02-22 10:30:27.531288', u'delta': u'0:00:00.008355', u'stderr': u'', u'rc': 0, 'stdout_lines': [], u'warnings': []}: template error while templating string: unexpected '.'. String: echo '{{ .NetworkSettings.IPAddress }}' > /dev/null"}
        to retry, use: --limit @/Users/meyers/ansible/ansible_bugs/args_template/main.retry

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=1
```

**Comment by pshchelo** [comment]
this might have to be related to https://github.com/ansible/ansible/issues/20098.

There was a bug in Jinja2 2.9.[0-4] that Ansible community fixed by hard-coding Jinja2<2.9 to PyPI releases 2.2.1.0 and 2.1.4.0 (before, there was no capping of Jinja2 version).

The bug was actually fixed in Jinja2==2.9.5, so the difference might be due to on Ansible 2.2.0.0 you get Jinja2==2.9.5 (or whatever latest version), but on 2.2.1.0 you get Jinja2==2.8.1 (or smth like that).

**Comment by evgeni** [comment]
Is this maybe the same as [CVE-2017-7466](https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2017-7466)?

Fixed in 2.2: https://github.com/ansible/ansible/commit/7ff9fa52cfcef2065f0db80d85dd94b9b754839c
Fixed in 2.3/devel: https://github.com/ansible/ansible/commit/0d418789a298561fded9bce977d34babc9097079

**Comment by sivel** [comment]
This issue was resolved in the v2.3.0 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
